doc/build-openbsd.md

```diff
@@ -63,50 +62,50 @@ export BDB_PREFIX="/path/to/bitcoin/depends/x86_64-unknown-openbsd"
 Bitcoin Core includes a GUI built with the cross-platform Qt Framework. To compile the GUI, Qt 5 is required.
 
 ```bash
-pkg_add qt5
+pkg_add qtbase qttools
 ```
 
-## Building Bitcoin Core
-
-**Important**: Use `gmake` (the non-GNU `make` will exit with an error).
+#### Test Suite Dependencies
+There is an included test suite that is useful for testing code changes when developing.
+To run the test suite (recommended), you will need to have Python 3 installed:
 
-Preparation:
 ```bash
+pkg_add install python  # Select the newest version of the package.
+```
 
-# Adapt the following for the version you installed (major.minor only):
-export AUTOCONF_VERSION=2.71
-export AUTOMAKE_VERSION=1.16
+## Building Bitcoin Core
 
-./autogen.sh
+Prepare a build directory:
```

Comment:
Aren't we using modern CMake invocations, where this isn't needed?

Reply:
I [think](https://github.com/hebasto/bitcoin/pull/167/files#r1579324545) this is a better option for the documentation purpose:

> The point is the `mkdir build && cd build` invocation guaranties the _clean_ build environment. Another benefit of `cd build` is that there is no need to specify the `--test-dir` option when running `ctest`.

Reply:
Right, but isn't the point of using a dir, that you can just blow it away, and guarantee a clean env? i.e I'll regularly just `rm -r build`, and then re-run CMake commands, without having to hop around in directories.

Reply:
I do the same :)

Or use the `--fresh` option.

But I believe that the documentation should be foolproof. It has to suggest the safest way to do things. Users will find their own ways that suit their needs better.

Reply:
> It has to suggest the safest way to do things.

Why is using mkdir and cd safer than using the CMake option that exists to do exactly the same thing?

Reply:
> I think it's because -B will reuse an existing dirty build directory

Sure, but there's nothing preventing a user who's hopping around dirs and always recreating them manually from accidently doing the exact same thing?

Reply:
You are right but in terms of documentation, how do we know that their build directory is empty? A mkdir sets the expectation that it was expected that this directory didn't already exist.

Reply:
That only works the first time, and nowhere do we tell users to cleanup and recreate their build dirs, so anyone who's already built once, could easily re-use the same build dir with different code etc.

Reply:
[Reworked](https://github.com/hebasto/bitcoin/pull/167#issuecomment-2077532788).

Reply:
`mkdir` ensures pristine build directory only if we assume that the user will stop if it gives an error. I can imagine some users continuing nevertheless (I have seen people that don't recognize error messages from normal diagnostic messages or just accidentally don't read that extra line on the screen which reads "mkdir: build: File exists").

If we want to stress on clean build directory then we need to start with a command like `rm -fr build` or `cmake --fresh`.